pyaedt/filtersolutions_core/optimization_goals_table.py

```diff
@@ -0,0 +1,411 @@
+            parameter_name_bytes_value,
+            weight_bytes_value,
+            enabled_bytes_value,
+        )
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def append_row(
+        self,
+        lower_frequency=None,
+        upper_frequency=None,
+        goal_value=None,
+        condition=None,
+        parameter_name=None,
+        weight=None,
+        enabled=None,
+    ):
+        """Append a new row of parameters into the optimization goals table.
+
+        Parameters
+        ----------
+        lower_frequency: str, optional
+            Lower frequency value to set.
+        upper_frequency: str, optional
+            Upper frequency value to set.
+        goal_value: str, optional
+            Goal value to set.
+        condition: str, optional
+            Condition value to set.
+        parameter_name: str, optional
+            Parameter name value to set.
+        weight: str, optional
+            Weight value to set.
+        enabled: str, optional
+            Enabled value to set.
+        """
+        if lower_frequency:
+            lower_frequency_bytes_value = bytes(lower_frequency, "ascii")
+        else:
+            lower_frequency_bytes_value = bytes("", "ascii")
+        if upper_frequency:
+            upper_frequency_bytes_value = bytes(upper_frequency, "ascii")
+        else:
+            upper_frequency_bytes_value = bytes("", "ascii")
+        if goal_value:
+            goal_value_bytes_value = bytes(goal_value, "ascii")
+        else:
+            goal_value_bytes_value = bytes("", "ascii")
+        if condition:
+            condition_bytes_value = bytes(condition, "ascii")
+        else:
+            condition_bytes_value = bytes("", "ascii")
+        if parameter_name:
+            parameter_name_bytes_value = bytes(parameter_name, "ascii")
+        else:
+            parameter_name_bytes_value = bytes("", "ascii")
+        if weight:
+            weight_bytes_value = bytes(weight, "ascii")
+        else:
+            weight_bytes_value = bytes("", "ascii")
+        if enabled:
+            enabled_bytes_value = bytes(enabled, "ascii")
+        else:
+            enabled_bytes_value = bytes("", "ascii")
+        status = self._dll.appendOptimizationGoalDefinitionRow(
+            lower_frequency_bytes_value,
+            upper_frequency_bytes_value,
+            goal_value_bytes_value,
+            condition_bytes_value,
+            parameter_name_bytes_value,
+            weight_bytes_value,
+            enabled_bytes_value,
+        )
+        pyaedt.filtersolutions_core._dll_interface().raise_error(status)
+
+    def insert_row(
+        self,
+        row_index,
+        lower_frequency=None,
+        upper_frequency=None,
+        goal_value=None,
+        condition=None,
+        parameter_name=None,
+        weight=None,
+        enabled=None,
+    ):
+        """Insert a new row of parameters into the optimization goals table.
```

Comment:
What happens to other rows at higher indices? What happens if there are already 50 entries in the table?